File: Week 01/id_063/LeetCode_283_063.java
/*
    就是简单的数组移动操作
*/


public class Solution {
    public void moveZeroes(int[] nums) {
        int idx1, idx2;

        for (idx1 = 0, idx2 = 0; idx2 < nums.length; idx2++) {
            if (nums[idx2] != 0) {
                nums[idx1++] = nums[idx2];
            }
        }

        while (idx1 < nums.length) {
            nums[idx1++] = 0;
        }
    }
}

File: Week 01/id_063/LeetCode_283_063.cpp
#include <vector>
using namespace std;

class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int idx_store = 0;

        for (int i = 0; i < nums.size(); i++) {
            if (nums[i]) {
                nums[idx_store++] = nums[i];
            }
        }

        fill(nums.begin() + idx_store, nums.end(), 0);
    }
};